### PR TITLE
Add playlist downloading functionality from BeastSaber

### PR DESCRIPTION
**Pull request by Parapass**
to test:
1. ctrl+shift+i
2. click any playlist button and make it bsaber:// instead of modsaber://
3. click the one click button
4. yay playlist downloaded 😄

### CODE REVIEW

File: ModAssistant/Classes/OneClickInstaller.cs [hunk 1]
```diff
@@ -72,7 +77,7 @@ private static async Task BeatSaver(Uri uri)
 
             if (BypassDownloadCounter)
             {
-                await DownloadAsset(BeatSaverURLPrefix + Response.directDownload, CustomSongsFolder, Response.hash + ".zip");
+                DownloadAsset(BeatSaverURLPrefix + Response.directDownload, CustomSongsFolder, Response.hash + ".zip");
```
Comment (Assistant) [inline ModAssistant/Classes/OneClickInstaller.cs]: Why this?

Comment (Parapass) [reply ModAssistant/Classes/OneClickInstaller.cs]: Forgot to edit out. Sorry.

File: ModAssistant/Classes/OneClickInstaller.cs [hunk 2]
```diff
@@ -72,7 +77,7 @@ private static async Task BeatSaver(Uri uri)
 
             if (BypassDownloadCounter)
             {
-                await DownloadAsset(BeatSaverURLPrefix + Response.directDownload, CustomSongsFolder, Response.hash + ".zip");
+               await DownloadAsset(BeatSaverURLPrefix + Response.directDownload, CustomSongsFolder, Response.hash + ".zip");
```
Comment (Assistant) [inline ModAssistant/Classes/OneClickInstaller.cs]: Why this?

Comment (Parapass) [reply ModAssistant/Classes/OneClickInstaller.cs]: it gone.

**Comment by Assistant** [comment]
I'm trying to get some people to look at this so they can see if it needs changes, and if not implement it on BeastSaber's side

File: ModAssistant/Pages/Options.xaml.cs [hunk 3]
```diff
@@ -120,12 +122,22 @@ public void BeatSaverProtocolHandler_Checked(object sender, RoutedEventArgs e)
         {
             OneClickInstaller.Register("beatsaver");
         }
-
+
```
Comment (Assistant) [inline ModAssistant/Pages/Options.xaml.cs]: Make sure to clean up this line too

Comment (Parapass) [reply ModAssistant/Pages/Options.xaml.cs]: will do

**Comment by Parapass** [comment]
Error: Changing it to bsaber:// and downloading a .bplist file instead of a .json results in the page being downloaded instead of the playlist.

**Comment by Parapass** [comment]
Error was fixed. Now to debug why the songs aren't downloading (╯°□°）╯︵ ┻━┻

**Comment by Parapass** [comment]
Converted to draft as I'm still working on this PR.

**Comment by Parapass** [comment]
Closing because new changes.

**Pull request closed without merging**